traceon/geometry.py

```diff
@@ -673,6 +670,126 @@ def aperture(height, radius, extent, z=0.):
         Path"""
         return Path.line([extent, 0., -height/2], [radius, 0., -height/2])\
                 .line_to([radius, 0., height/2]).line_to([extent, 0., height/2]).move(dz=z)
+
+    @staticmethod
+    def arc_polar(radius, angle, start=[0,0,1], plane_normal=[0,1,0], direction=[1,0,0]):
+        """Return an arc of specified by polar coordinates. The arc lies in a plane defined by the 
+        provided normal vector and curves from the start point in the specified direction 
+        counterclockwise around the normal.
+
+        Parameters
+        ---------------------------
+        radius : float
+            The radius of the arc.
+        angle : float
+            The angle subtended by the arc (in radians)
+        start: (3,) float
+            The start point of the arc
+        plane_normal : (3,) float
+            The normal vector of the plane containing the arc
+        direction : (3,) float
+            A tangent of the arc at the starting point. 
+            Must lie in the specified plane. Does not need to be normalized. 
+        Returns
+        ----------------------------
+        Path"""
+        start = np.array(start, dtype=float)
+        plane_normal = np.array(plane_normal, dtype=float)
+        direction = np.array(direction, dtype=float)
+
+        dot_product = np.dot(direction, plane_normal)
+
+        if not np.isclose(dot_product, 0, atol=1e-7):
+            corrected_direction = direction - (dot_product / np.linalg.norm(plane_normal)**2) * plane_normal
+            raise AssertionError(
+                f"The provided direction {direction} does not lie in the specified plane. \n"
+                f"The closed valid direction is {np.round(corrected_direction, 10)}.")
+        
+        direction /= np.linalg.norm(direction)
+        plane_normal /= np.linalg.norm(plane_normal)
+
+        if angle < 0:
+            direction, angle = -direction, -angle
+
+        center = start - radius * np.cross(direction, plane_normal)
+        center_to_start = start - center
+        
+        def f(l):
+            theta = l/radius
+            return center + np.cos(theta) * center_to_start + np.sin(theta)*np.cross(plane_normal, center_to_start)
+        
+        return Path(f, radius*angle)
+    
+    def arc_to_polar(self, radius, angle, plane_normal=[0, 1, 0]):
```

Comment:
I think we have to think about the naming a bit. `to` was used in `line_to(point)` because it reads as `line to point` which makes sense. Here `arc_to_polar` doesn't sound as nice I think. Maybe `continue_with` would be the good naming since we follow the tangent vector `.continue_with_polar_arc(...)`. What do you think?

Reply:
what about `extend_with_polar_arc`? I would suggest to rename `arc_to` to `extend_with_arc` then as well for consistency.

Reply:
Strictly speaking this is a point release (v0.7.2 -> v0.7.3) so we are not allowed to break already existing functions.. you can however create a second method `extend_with_arc` and then remove arc_to from the documentation `__pdoc__['arc_to'] = False`. I think we should then have to do the same with `line_to` for consistency: `line_to` -> `extend_with_line`

Reply:
agreed. Should i add a deprecation warning that these functions will be removed in the first non-backwards compatible version (I guess v0.8.0?)

Reply:
Yes, maybe this would be a good application of the `warn` function: https://docs.python.org/3/library/warnings.html#warnings.warn